test/multiple-user-principal-types.test.js

```diff
@@ -0,0 +1,485 @@
+      app.model(Role, {dataSource: 'db'});
+
+      RoleMapping = app.registry.getModel('RoleMapping');
+      app.model(RoleMapping, {dataSource: 'db'});
+    });
+
+    describe('role/user relations:', function() {
+      beforeEach(function(done) {
+        OneUser.create({name: 'Eric', email: 'x@y.com', password: 'foobar'})
+        .then(function(u) {
+          user = u;
+          return Role.create({name: 'userRole'});
+        })
+        .then(function(role) {
+          userRole = role;
+          done();
+        })
+        .catch(done);
+      });
+
+      it('role.users() should return users when using custom user principalType',
+      function(done) {
+        userRole.principals.create({principalType: OneUser.modelName, principalId: user.id})
+        .then(function() {
+          return userRole.users({where: {principalType: OneUser.modelName}});
+        })
+        .then(function(users) {
+          assert.equal(users.length, 1);
+          assert.equal(users[0].id, user.id);
+          done();
+        })
+        .catch(function(err) {
+          done(err);
+        });
+      });
+
+      it('role.users() should return empty array when using invalid principalType',
+      function(done) {
+        userRole.principals.create({principalType: 'invalidModelName', principalId: user.id})
+        .then(function() {
+          return userRole.users({where: {principalType: 'invalidModelName'}});
+        })
+        .then(function(users) {
+          assert.equal(users.length, 0);
+          done();
+        })
+        .catch(function(err) {
+          done(err);
+        });
+      });
+
+      it('principal.user() should return user', function(done) {
+        userRole.principals.create({principalType: OneUser.modelName, principalId: user.id})
+        .then(function(principal) {
+          return principal.user();
+        })
+        .then(function(res) {
+          assert(res);
+          assert.equal(res.id, user.id);
+          done();
+        })
+        .catch(done);
+      });
+
+      it('principal.user() should return null when created with invalid principalType',
+      function(done) {
+        userRole.principals.create({principalType: 'invalidModelName', principalId: user.id})
+        .then(function(principal) {
+          return principal.user();
+        })
+        .then(function(res) {
+          assert(!res);
+          done();
+        })
+        .catch(done);
+      });
+    });
+
+    describe('isInRole() & getRole():', function(done) {
+      var user, role;
+
+      beforeEach(function(done) {
+        OneUser.create({name: 'Eric', email: 'x@y.com', password: 'foobar'})
+          .then(function(u) {
+            user = u;
+            return Role.create({name: 'userRole'});
+          })
+          .then(function(r) {
+            role = r;
+            return role.principals.create({principalType: OneUser.modelName,
+              principalId: user.id});
+          })
```

Comment:
reused already existing users.
about userRole : thought you might want to avoid the extra burden of creating it for every tests, but... done